control-plane/agents/src/bin/core/node/wrapper.rs

```diff
@@ -1377,6 +1378,32 @@ impl NexusChildActionApi<NexusChildActionContextNode> for Arc<tokio::sync::RwLoc
     }
 }
 
+#[async_trait]
+impl SnapshotApi<Nexus> for Arc<tokio::sync::RwLock<NodeWrapper>> {
+    type CreateIn = CreateNexusSnapshot;
+    type CreateOutput = CreateNexusSnapshotResp;
+
+    async fn create_snapshot(
+        &self,
+        _request: &Self::CreateIn,
+    ) -> Result<Self::CreateOutput, SvcError> {
+        todo!()
+    }
+}
+
+#[async_trait]
+impl SnapshotApi<Replica> for Arc<tokio::sync::RwLock<NodeWrapper>> {
```

Comment:
Even so, it becomes quite difficult to call this... I'll push another commit where we split this into 2 traits.

Reply:
I've pushed an alternative approach with 2 snapshot traits with different method names, allowing us to call them in a nicer way, example:
```
node.create_repl_snapshot(...) vs
SnapshotApi::<Replica>::create_snapshot(&node)
```
Let me know your thoughts @dsharma-dc @Abhinandan-Purkait

Reply:
Agree about implementing for RpcClient.
The first approach (b10d9f44e517b4d3a0c71b1472856ee8376944e4) is exactly how I intended it in the beginning, but couldn't figure out generic parameters definition owing to my language awareness barrier. :) 
What is the difficulty that you mention in calling this? Is it about the using full templated name to call?
Second approach is a bit more scattered though.
Another question: Is it mandatory to impl the trait for node wrapper, irrespective of approach?

Reply:
The difficulty is that you can't just call it directly, 
`node.create_snapshot()`
 on a module which imports both traits, because the compiler can't figure out which one to call. You have to disambiguate like so:
`SnapshotApi::<Replica>::create_snapshot(&node)`
which makes it a bit complicated to use..